SOS/APP/app.c

```diff
@@ -0,0 +1,231 @@
+/*
+* app.c
+*
+* Created: 5/30/2023 2:25:22 PM
+*  Author: Alaa
+*/
+
+#include "interrupts.h"
+
+#include "led_interface.h"
+#include "btn_interface.h"
+#include "sos_interface.h"
+
+#include "app.h"
+
+static void app_led0(void);
+static void app_led1(void);
+static void app_start_btn(void);
+static void app_stop_btn(void);
+
+static str_btn_config_t_ gl_str_start_btn;
+static str_btn_config_t_ gl_str_stop_btn;
+
+static str_sos_task_t_ gl_str_led0_task;
+static str_sos_task_t_ gl_str_led1_task;
+static str_sos_task_t_ gl_str_stop_btn_task;
+
+static enu_btn_state_t_ gl_enu_start_btn_state = BTN_STATE_NOT_PRESSED;
+static enu_btn_state_t_ gl_enu_stop_btn_state = BTN_STATE_NOT_PRESSED;
+
+
+enu_app_init_status_t_ app_init(void)
+{
+	
+	/* Initialize LEDs */
+	enu_led_error_t_ lo_enu_led_status = 0;
+	lo_enu_led_status = led_init(APP_LED_0_PORT, APP_LED_0_PIN);
+	lo_enu_led_status += led_init(APP_LED_1_PORT, APP_LED_1_PIN);
+	
+	if(lo_enu_led_status != LED_OK) {return APP_INIT_FAILED;}

```

Comment:
Return Type Check Note:
This check shall be added between the led_init(APP_LED_0_PORT, APP_LED_0_PIN); && led_init(APP_LED_1_PORT, APP_LED_1_PIN);
As if LED0 Initialization Failed. Then why you may need to initalize LED1?